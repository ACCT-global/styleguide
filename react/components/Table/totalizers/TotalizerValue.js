import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import config from 'vtex-tachyons/config.json'

import Spinner from '../../Spinner'

class TotalizerValue extends PureComponent {
  render() {
    const {
      item: { value, isLoading },
    } = this.props

    if (isLoading) {
      return (
        <div className="mt2">
          <Spinner
            size="14"
            color={config.semanticColors.background['base--inverted']}
          />
        </div>
      )
    }

    if (value) {
      return <div className="f4 fw5 c-on-base">{value}</div>
    }

    return null
  }
}

TotalizerValue.propTypes = {
  item: PropTypes.object,
}

export default TotalizerValue
