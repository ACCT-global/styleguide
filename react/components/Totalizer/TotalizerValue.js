import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

import Spinner from '../Spinner'

class TotalizerValue extends PureComponent {
  render() {
    const {
      item: { value, isLoading },
      mobileScroll,
      testId,
    } = this.props

    if (isLoading) {
      return (
        <div data-testid={`${testId}__loading`} className="c-muted-1">
          <Spinner size={14} color="currentColor" />
        </div>
      )
    }

    if (value == null) {
      return null
    }

    return (
      <div
        data-testid={testId}
        className={`f4 fw5 c-on-base ${mobileScroll ? 'nowrap' : ''}`}
      >
        {value}
      </div>
    )
  }
}

TotalizerValue.propTypes = {
  mobileScroll: PropTypes.bool,
  item: PropTypes.shape({
    value: PropTypes.node,
    isLoading: PropTypes.bool,
  }),
  testId: PropTypes.string,
}

export default TotalizerValue
