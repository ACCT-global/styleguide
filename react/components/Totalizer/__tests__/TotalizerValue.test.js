import React from 'react'
import { render } from 'react-testing-library'

import TotalizerValue from '../TotalizerValue'

describe('Totalizer/Value spec', () => {
  const testId = 'testing_id'
  const value = 'value'

  it('should match snapshot on desktop', () => {
    const Value = render(<TotalizerValue item={{ value }} testId={testId} />)
    expect(Value.asFragment()).toMatchSnapshot()
  })

  it('should match snapshot on mobile', () => {
    const Value = render(
      <TotalizerValue item={{ value }} testId={testId} mobileScroll />
    )
    expect(Value.asFragment()).toMatchSnapshot()
  })

  it('should match snapshot while loading', () => {
    const Value = render(
      <TotalizerValue item={{ value, isLoading: true }} testId={testId} />
    )
    expect(Value.asFragment()).toMatchSnapshot()
  })

  it('should match snapshot while loading on mobile', () => {
    const Value = render(
      <TotalizerValue
        item={{ value, isLoading: true }}
        testId={testId}
        mobileScroll
      />
    )
    expect(Value.asFragment()).toMatchSnapshot()
  })

  it('should render correctly', () => {
    const Value = render(<TotalizerValue item={{ value }} />)
    const ElementToTest = Value.getByTestId(testId)
    expect(ElementToTest).toBeTruthy()
    expect(ElementToTest.childElementCount).toBe(0)
    expect(ElementToTest.textContent).toBe(value)
  })

  it('should render correctly while loading', () => {
    const Loading = render(<TotalizerValue item={{ value, isLoading: true }} />)
    const ElementLoading = Loading.getByTestId(`${testId}__loading`)
    expect(ElementLoading).toBeTruthy()
    expect(ElementLoading.childElementCount).toBe(1)
    expect(ElementLoading.textContent).not.toBe(value)
  })
})
