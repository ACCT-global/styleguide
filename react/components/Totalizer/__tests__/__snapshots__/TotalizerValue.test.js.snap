// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Totalizer/Value spec should match snapshot on desktop 1`] = `
<DocumentFragment>
  <div
    class="f4 fw5 c-on-base "
    data-testid="testing_id"
  >
    value
  </div>
</DocumentFragment>
`;

exports[`Totalizer/Value spec should match snapshot on mobile 1`] = `
<DocumentFragment>
  <div
    class="f4 fw5 c-on-base nowrap"
    data-testid="testing_id"
  >
    value
  </div>
</DocumentFragment>
`;

exports[`Totalizer/Value spec should match snapshot while loading 1`] = `
<DocumentFragment>
  <div
    class="c-muted-1"
    data-testid="testing_id__loading"
  >
    <svg
      class="vtex__icon-spinner   "
      height="14"
      preserveAspectRatio="xMidYMid"
      viewBox="0 0 100 100"
      width="14"
      xmlns="http://www.w3.org/2000/svg"
    >
      <style>
        
            @keyframes vtex-spinner-rotate {
              from {
                transform-origin: 50% 50%;
                transform: rotate(0deg);
              }
              to {
                transform-origin: 50% 50%;
                transform: rotate(360deg);
              }
            }

            @keyframes vtex-spinner-fill {
              0% {
                stroke-dasharray: 0 0 2 251.32741228718345;
              }
              50% {
                stroke-dasharray: 0 0 188.49555921538757 251.32741228718345;
              }
              100% {
                stroke-dasharray: 0 249.32741228718345 188.49555921538757 251.32741228718345;
              }
            }

            .vtex-spinner_circle {
              animation: vtex-spinner-fill 1.25s infinite cubic-bezier(0.455, 0.030, 0.515, 0.955), vtex-spinner-rotate 0.625s infinite linear;
            }
          
      </style>
      <circle
        class="vtex-spinner_circle"
        cx="50"
        cy="50"
        fill="none"
        r="40"
        stroke="currentColor"
        stroke-dasharray="0 0 188.49555921538757 251.32741228718345"
        stroke-dashoffset="1"
        stroke-linecap="round"
        stroke-width="10"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`Totalizer/Value spec should match snapshot while loading on mobile 1`] = `
<DocumentFragment>
  <div
    class="c-muted-1"
    data-testid="testing_id__loading"
  >
    <svg
      class="vtex__icon-spinner   "
      height="14"
      preserveAspectRatio="xMidYMid"
      viewBox="0 0 100 100"
      width="14"
      xmlns="http://www.w3.org/2000/svg"
    >
      <style>
        
            @keyframes vtex-spinner-rotate {
              from {
                transform-origin: 50% 50%;
                transform: rotate(0deg);
              }
              to {
                transform-origin: 50% 50%;
                transform: rotate(360deg);
              }
            }

            @keyframes vtex-spinner-fill {
              0% {
                stroke-dasharray: 0 0 2 251.32741228718345;
              }
              50% {
                stroke-dasharray: 0 0 188.49555921538757 251.32741228718345;
              }
              100% {
                stroke-dasharray: 0 249.32741228718345 188.49555921538757 251.32741228718345;
              }
            }

            .vtex-spinner_circle {
              animation: vtex-spinner-fill 1.25s infinite cubic-bezier(0.455, 0.030, 0.515, 0.955), vtex-spinner-rotate 0.625s infinite linear;
            }
          
      </style>
      <circle
        class="vtex-spinner_circle"
        cx="50"
        cy="50"
        fill="none"
        r="40"
        stroke="currentColor"
        stroke-dasharray="0 0 188.49555921538757 251.32741228718345"
        stroke-dashoffset="1"
        stroke-linecap="round"
        stroke-width="10"
      />
    </svg>
  </div>
</DocumentFragment>
`;
